mycpp/cppgen_pass.py

```diff
@@ -534,14 +547,28 @@ def visit_yield_from_expr(self, o: 'mypy.nodes.YieldFromExpr') -> T:
         pass
 
     def visit_yield_expr(self, o: 'mypy.nodes.YieldExpr') -> T:
-        pass
+        assert self.current_func_node in self.yield_accumulators
+        self.write_ind('%s->append(', self.yield_accumulators[self.current_func_node][0])
+        self.accept(o.expr)
+        self.write(');\n')
 
     def _WriteArgList(self, o):
       self.write('(')
       for i, arg in enumerate(o.args):
         if i != 0:
           self.write(', ')
         self.accept(arg)
+
+      # Will be set if we're:
+      # a) accumulating the output of an iterator
+      # b) constructing an iterator with the result of (a)
+      if self.current_stmt_node in self.yield_accumulators:
+          if len(o.args) > 0:
```

Comment:
Indentation looks a little inconsistent here

I know it's already messed up because MyPy uses 4 and I use 2 ... at some point we should run a formatter over it